app/helpers/patient_query_helper.rb

```diff
@@ -282,6 +282,24 @@ def advanced_filter(patients, filters, tz_offset)
         patients = advanced_filter_relative_date(patients, :symptom_onset, filter, tz_diff, :date)
       when 'continous-exposure'
         patients = patients.where(continuous_exposure: filter[:value].present? ? true : [nil, false])
+      when 'close-contact-with-known-case'
+        if filter[:value].blank?
+          patients = patients.where(contact_of_known_case_id: [nil, ''])
+        else
+          value_array = filter[:value].split(/\s*,\s*/)
+          filtered_patient_ids = []
+          case filter[:additionalFilterOption]
+          when 'Exact Match'
+            value_array.each do |value|
+              filtered_patient_ids += patients.where('contact_of_known_case_id REGEXP ?', "(^|,)(#{value})(,|$)").pluck(:id)
```

Comment:
Looking at this again, I think we can simplify this even further so we're not making n queries where n is the number of comma-separated values. This is really just an OR, but we're doing a pluck for every value right now which means a query is being executed num values times. Instead, we could just build a single query and do a pluck at the end so we only reach out to the db once. There's a bunch of ways to do this, the cleanest I can think of is taking advantage of REGEXP again. Ex: `first_name REGEXP 'kylie|sarah'`. This way you can build the appropriate REGEXP ahead of time using the `values_array` and the only pluck once. And we can apply the same concept below, though since it's not REGEXP we can build the query string with LIKE the same way I think.

Reply:
After some playing around, I was able to use REGEXP for both.  It's much cleaner and more efficient!

Reply:
See here: https://github.com/SaraAlert/SaraAlert/pull/701/commits/1f5223888522e4e3b0d467ff131d8474f821a743